src/app/header/header.component.ts

```diff
@@ -1,20 +1,33 @@
-import { Component, Input } from '@angular/core';
-import { AuthenticationService } from '../shared/services/authentication.service';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
 import { Router } from '@angular/router';
 
+import { AuthenticationService } from '../shared/services/authentication.service';
+
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.css']
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit, OnDestroy {
+  userName: string = '';
+
   constructor(private authService: AuthenticationService, private router: Router) {}
 
-  @Input() userName: string = '';
+  ngOnInit(): void {
+    this.authService.userSubject.subscribe(
+      value => {
+        this.userName = value.name.first;
+      }
+    )
+  }
 
   onLogOut(): void {
     this.authService.logout();
     this.userName = '';
     this.router.navigate(['/login']);
   }
+
+  ngOnDestroy(): void {
+    this.authService.userSubject.unsubscribe();
```

Comment:
You have unsubscribed a little bit wrong
You have to create the variable and then unsubscribe from it
Please read more about it here
[6-ways-to-unsubscribe-from-observables-in-angular](https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f)

Reply:
done, with Subscription variable